R/florabank_queries.R

```diff
@@ -0,0 +1,424 @@
+#' Query the florabank to get taxon trait values for (a) taxon trait(s)
+#'
+#' This function takes as input (part of) a taxon trait, queries the florabank
+#' and returns the taxon trait values in a tidy data format
+#'
+#' @param connection A connection to the florabank database. See the example section
+#' for how to connect and disconnect to the database.
+#'
+#' @param trait_name A (part of) a trait name for which you want to get the
+#' associated taxon-specific trait values. If this is missing, the function
+#' returns an error and prints a message showing all possible trait names.
+#'
+#' @return A dataframe containing the trait values for each species and for all
+#' partially matched traits. The dataframe contains the variables TaxonID,
+#' TaxonAfkorting, TaxonWetenschappelijk, TaxonNederlands, Kenmerk, Code,
+#' Omschrijving en Rekenwaarde. The first four variables identify the taxon,
+#' the latter four variables relate to the taxon traits.
+#'
+#' @importFrom dplyr
+#' tbl
+#' collect
+#' distinct
+#' pull
+#' %>%
+#' inner_join
+#' left_join
+#' filter
+#' select
+#' rename
+#' @importFrom rlang .data
+#' @importFrom assertthat assert_that
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # connect to florabank
+#' db_connectie <- connect_inbo_dbase("D0021_00_userFlora")
+#'
+#' # get all Ellenberg values via partial matching
+#' fb_ellenberg <- florabank_traits(db_connectie, "llenberg")
+#'
+#' # get all red lists via partial matching
+#' fb_rodelijsten <- florabank_traits(db_connectie, "rode")
+#'
+#' # get only the red list for vascular plant species
+#' fb_rodelijstvaatplanten <- florabank_traits(db_connectie, "Rode lijst Vaatplanten")
+#'
+#' #if the trait_name argument is missing, a list of possible names is printed
+#' florabank_traits(db_connectie)
+#'
+#' #disconnect from florabank
+#' dbDisconnect(db_connectie)
+#' }
+
+florabank_traits <- function(connection, trait_name) {
+
+  assert_that(class(connection)[1] == "Microsoft SQL Server",
+              msg = "Not a connection object to database.")
+  assert_that(connection@info$dbname == "D0021_00_userFlora")
+
+  if (missing(trait_name)) {
+    traitnames <- tbl(connection, "tblTaxonKenmerk") %>%
+      distinct(.data$Naam) %>%
+      collect() %>%
+      pull(.data$Naam)
+    message(paste(traitnames, collapse = ", "))
```

Comment:
It is better to incorporate the messing into the error. In some case the error will be visible, while the message isn't.

Reply:
Resolved in a563434